pkg/plugins/connection/ssh.go

```diff
@@ -40,13 +45,59 @@ func (t *SSHOpen) Metadata() *gornir.TaskMetadata {
 	return t.Meta
 }
 
+type authMethodResult struct {
+	authMethods []ssh.AuthMethod
+	err         error
+}
+
+func getAuthMethods(host *gornir.Host, logger gornir.Logger) (*[]ssh.AuthMethod, error) {
+	var authMethods = []ssh.AuthMethod{ssh.Password(host.Password)}
+	// GetStrict should return a default value per `man ssh_config` if this fails, it is because we couldn't parse the config file
+	sshPrivKeyFname, err := ssh_config.GetStrict(host.Hostname, "IdentityFile")
+	if err != nil {
+		return nil, err
+	}
+	signer, err := getPubKeySigner(host, sshPrivKeyFname, logger)
+	// Drop private key auth from the list and fallback to user/pass
+	if err != nil {
+		return &authMethods, nil
+	}
+	authMethods = append(authMethods, ssh.PublicKeys(*signer))
+	return &authMethods, nil
+}
+
+func getPubKeySigner(host *gornir.Host, sshPrivKeyFname string, logger gornir.Logger) (*ssh.Signer, error) {
+	usr, _ := user.Current()
```

Comment:
I have no idea under which circumstances this may fail (and the docs don't specify) but I'd prefer not to ignore the errorrs

Reply:
Fair point - probably on non "typical" system setups/OS's and we should be explicit about that.  I'll fix.